### PR TITLE
indlib: 1.8.9 -> 1.9.0

### PR DESCRIPTION
**Pull request by hjones2199**
###### Motivation for this change
Support more astro device drivers and receive upstream bugfixes.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hjones2199** [comment]
Result of `nixpkgs-review pr 120459` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>indi-full</li>
    <li>indilib</li>
    <li>kstars</li>
    <li>phd2</li>
  </ul>
</details>

**Comment by hjones2199** [comment]
Result of `nixpkgs-review pr 120459` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>indi-full</li>
    <li>indilib</li>
    <li>kstars</li>
    <li>phd2</li>
  </ul>
</details>

File: pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix [hunk 1]
```diff
@@ -18,39 +17,51 @@
...
+    libjpeg libgphoto2 libraw libftdi1 libdc1394 ffmpeg fftw
+  ]
+  ++ lib.optionals withFirmware [
+    firmware
+  ];
+
+  postPatch = ''
+    substituteInPlace indi-qsi/CMakeLists.txt --replace "/lib/udev/rules.d" "lib/udev/rules.d"
+    substituteInPlace indi-armadillo-platypus/CMakeLists.txt --replace "/lib/udev/rules.d" "lib/udev/rules.d"
+    substituteInPlace indi-dsi/CMakeLists.txt --replace "/etc/udev/rules.d" "lib/udev/rules.d"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix:44]: That could also use for f in; do ... done.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -15052,7 +15052,6 @@ in
   indicator-application-gtk3 = callPackage ../development/libraries/indicator-application/gtk3.nix { };
 
   indilib = callPackage ../development/libraries/science/astronomy/indilib { };
-  indi-3rdparty = callPackage ../development/libraries/science/astronomy/indilib/indi-3rdparty.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:15055]: Do we need an alias for this?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120459` at 1f024e21 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>kstars</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/1b19a73aaca1c0358fead47f8ade3416">indi-full</a></li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>indilib</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (hjones2199) [reply pkgs/top-level/all-packages.nix:15055]: I don't think so. indi-3rdparty is only useful in the context of being wrapped into an environment with indilib, a use-case covered by indi-full. Same goes for indi-firmware.

**Comment by hjones2199** [comment]
I don't have an aarch64 system to test on, but I assume one of the indi-firmware drivers did it. If so, it should build now, I made indi-full only wrap indi-firmware into the environment on x86_64

File: pkgs/development/libraries/science/astronomy/indilib/indi-full.nix [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+    version = indi-version;
+    src = indi-3rdparty-src;
+    withFirmware = stdenv.isx86_64;
+    firmware = indi-firmware;
+  };
 in
-callPackage indi-with-drivers {
-  pkgName = "indi-full";
+callPackage ./indi-with-drivers.nix {
+  name = "indi-full-${indi-version}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/astronomy/indilib/indi-full.nix:23]:
```suggestion
  pname = "indi-full";
  version = indi-version;
```

**Pull request merged**